galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/ResultNames.java

```diff
@@ -0,0 +1,76 @@
+package dev.galasa.framework.api.ras.internal;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.validation.constraints.NotNull;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ServiceScope;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import dev.galasa.framework.spi.IFramework;
+import dev.galasa.framework.spi.IResultArchiveStoreDirectoryService;
+import dev.galasa.framework.spi.ResultArchiveStoreException;
+
+@Component(service = Servlet.class, scope = ServiceScope.PROTOTYPE, property = {
+"osgi.http.whiteboard.servlet.pattern=/ras/resultnames" }, name = "Galasa Test Result Names microservice")
+public class ResultNames extends HttpServlet {
+
+	private static final long serialVersionUID = 1L;
+
+	private static @NotNull List<String> resultsList = new ArrayList<>();
```

Comment:
No way should this be a field or even static.   This will clash with 2 of these servlets running exactly at the same time.   Needs to be a variable in the doGet